### PR TITLE
This PR adds test for the methods DQ_SerialWholeBody::{raw_fkm, fkm}

### PR DESCRIPTION
**Pull request by juanjqo**
-passing-passing)

Hi @mmmarinho,

This PR adds tests for the methods `DQ_SerialWholeBody::{raw_fkm, fkm}`. The motivation behind this PR is to test the methods added in this [PR](https://github.com/dqrobotics/python/pull/55). There are more methods to be added to those tests (e.g., pose_jacobian), which I'm planning to add in future PR's.

Best regards, 

Juancho

### CODE REVIEW

File: DQ_Kinematics_test.py [hunk 1]
```diff
@@ -16,43 +16,68 @@
...
+from dqrobotics.robot_modeling import DQ_SerialWholeBody
 
 # The data from MATLAB
 mat = scipy.io.loadmat('DQ_Kinematics_test.mat')
 # A list of random DQs
 h_list = get_list_of_dq_from_mat('random_dq_a', mat)
 # A list of random joint values
 serial_manipulator_q_list = get_list_of_vector_from_mat('random_q', mat)
+
+# A lists for the DQ_SerialWholeBody class
```
Comment (mmmarinho) [inline DQ_Kinematics_test.py]: @juanjqo Fix typo here

File: DQ_Kinematics_test.py [hunk 2]
```diff
@@ -16,43 +16,68 @@
...
+serial_whole_body_q_list    = get_list_of_vector_from_mat('random_q_serial_whole_body', mat)
+serial_whole_body_pose_list = get_list_of_dq_from_mat('result_of_serial_whole_body_fkm', mat)
+serial_whole_body_raw_pose_list = get_list_of_dq_from_mat('result_of_serial_whole_body_raw_fkm', mat)
+serial_whole_body_pose_jacobian_list = get_list_of_matrices_from_mat('result_of_serial_whole_body_jacobian', mat)
```
Comment (mmmarinho) [inline DQ_Kinematics_test.py]: @juanjqo Is this being used?

**Review by mmmarinho — CHANGES_REQUESTED** [review]
@juanjqo I suppose other fixes are dependent on this, but a couple of comments for now.

File: DQ_Kinematics_test.py (hunk 1 above)
Comment (juanjqo) [reply DQ_Kinematics_test.py]: @mmmarinho fixed!

File: DQ_Kinematics_test.py (hunk 2 above)
Comment (juanjqo) [reply DQ_Kinematics_test.py]: @mmmarinho no, I forgot to remove the parts related to the test of the `pose_jacobian` method. I removed it now.

**Review by mmmarinho — APPROVED** [review]
@juanjqo Seems ok. Does this implement all the test coverage you need?

**Comment by juanjqo** [comment]
@mmmarinho Concerning the data from Matlab, this PR generates data to test the methods `raw_fkm`, `fkm`, and `pose_jacobian` but does not implement tests for the latter (because of the bug reported [here](https://github.com/dqrobotics/cpp/issues/61)). Furthermore, there is no data generation for methods related to the pose jacobian derivatives yet. Besides those points, I implemented all the test coverage I needed, which corresponds to the following tests:

```python
    # DQ_SerialWholeBody.fkm
    def test_serial_whole_body_fkm(self):
        for q, x in zip(serial_whole_body_q_list, serial_whole_body_pose_list):
            self.assertEqual(serial_whole_body_robot.fkm(q), x, "Error in DQ_SerialWholeBody.fkm")

    # DQ_SerialWholeBody.raw_fkm
    def test_serial_whole_body_raw_fkm(self):
        for q, x in zip(serial_whole_body_q_list, serial_whole_body_raw_pose_list):
            self.assertEqual(serial_whole_body_robot.raw_fkm(q), x, "Error in DQ_SerialWholeBody.raw_fkm")
```

**Pull request merged**